src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.cs

```diff
@@ -143,5 +143,15 @@ await FileStreamHelpers
                 }
             }
         }
+
+        public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback? callback, object? state) =>
+            TaskToApm.Begin(ReadAsyncInternal(new Memory<byte>(buffer, offset, count)).AsTask(), callback, state);
+
+        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback? callback, object? state) =>
+            TaskToApm.Begin(WriteAsyncInternal(new ReadOnlyMemory<byte>(buffer, offset, count)).AsTask(), callback, state);
+
+        public override int EndRead(IAsyncResult asyncResult) => TaskToApm.End<int>(asyncResult);
+
+        public override void EndWrite(IAsyncResult asyncResult) => TaskToApm.End(asyncResult);
```

Comment:
Nit: Can you reorganize this so that BeginWrite/EndWrite are next to each other and up with the other writing methods, and then similarly for BeginRead/EndRead?

Reply:
I had followed the order of the APM methods on `Net5CompatFileStreamStrategy`, but fixed it now.